sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogUtils.scala

```diff
@@ -169,9 +169,15 @@ object ExternalCatalogUtils {
       spec1: TablePartitionSpec,
       spec2: TablePartitionSpec): Boolean = {
     spec1.forall {
+      case (partitionColumn, null | DEFAULT_PARTITION_NAME) =>
```

Comment:
Can we add a util function `isInvalidPartitionValue`? then the code can be
```
case (partitionColumn, value) if isInvalidPartitionValue(value) =>
  isInvalidPartitionValue(spec2(partitionColumn))
```

Reply:
will we hit empty string partition value here?

Reply:
> Can we add a util function isInvalidPartitionValue?

1. Why are partition values invalid? They are still valid here
2. Where else will the function be used. Since this is only the place, wouldn't be better to keep the code embedded here?

> will we hit empty string partition value here?

Empty string is handling earlier. We cannot have it here.

Reply:
how about `isNullPartitionValue`